### PR TITLE
Quality Assurance QA 

### PR DESCRIPTION
**Pull request by Rytiggy**
### Changes
- Fix metadata pages not saving right when entering data 
- Fix fuzzy match always firing even when the user had data entered
- Fix table displaying info on grant detail page
- Fix date on grant detail page
- Fix submission detail page not showing the right info.

### CODE REVIEW

File: app/components/metadata-form.js [hunk 1]
```diff
@@ -281,6 +281,27 @@ function fuzzySet(arr, useLevenshtein, gramSizeLower, gramSizeUpper) {
 }
 
 
+// Not related to fuzzy lookup
+function hasDuplicates(array) {
```
[the remark this answers is not in the document]
Comment (Rytiggy) [reply app/components/metadata-form.js]: This was code i ment to remove good eye

[the remark this answers is not in the document]
Comment (cameronblandford) [reply app/components/metadata-form.js]: <3

File: app/components/metadata-form.js [hunk 2]
```diff
@@ -290,50 +311,62 @@ export default Ember.Component.extend({
...
-              } else if (doiInfo[doiEntry].length > 0) {
-                // Predicts data with .61 accuracy
-                if (f.get(doiEntry)[0][0] > 0.61) {
-                  console.log(doiEntry, doiInfo[doiEntry], f.get(doiEntry)[0][0]);
-                  // set the found record to the metadata
-                  prePopulateData[f.get(doiEntry)[0][1]] = doiInfo[doiEntry];
+    if(newForm.id){
+      // Populate form with data if there is any to populate with.
+      if (!metadata) {
+        console.log('if', metadata)
```
[the remark this answers is not in the document]
Comment (Rytiggy) [reply app/components/metadata-form.js]: :3

File: app/components/metadata-form.js [hunk 3]
```diff
@@ -344,19 +377,60 @@ export default Ember.Component.extend({
...
+
+
+            // console.log(hasDuplicates(metadata))
+            // metadata = removeDuplicates(metadata)
+            metadata.push({
+              id: formId,
+              data: value,
+            });
+            // remove any duplicates
+            let uniqIds = {}, source = metadata;
```
[the remark this answers is not in the document]
Comment (cameronblandford) [reply app/components/metadata-form.js]: OH, I GET IT

File: app/components/metadata-form.js [hunk 4]
```diff
@@ -344,19 +377,60 @@ export default Ember.Component.extend({
...
+            let filtered = source.reverse().filter(obj => !uniqIds[obj.id] && (uniqIds[obj.id] = true));
```
[the remark this answers is not in the document]
Comment (cameronblandford) [reply app/components/metadata-form.js]: Nevermind, I get it now! Cool way of handling it.

File: app/components/metadata-form.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Rytiggy) [reply app/components/metadata-form.js]: What that line of code does is removes any duplicates that may have been generated from the user going back to edit a metadata form.  the UniqIds is used to keep track of which IDs are unique.

File: app/components/workflow-metadata.js [hunk 5]
```diff
@@ -287,7 +287,6 @@ export default Component.extend({
   currentFormStep: 0,
 
   didInsertElement() {
-    debugger;
```
[the remark this answers is not in the document]
Comment (Rytiggy) [reply app/components/workflow-metadata.js]: 😀

**Pull request merged**